gcloud-java-storage/src/main/java/com/google/gcloud/storage/BlobReadChannelImpl.java

```diff
@@ -148,4 +144,116 @@ public int read(ByteBuffer byteBuffer) throws IOException {
     }
     return toWrite;
   }
+
+  static class StateImpl implements RestorableState<BlobReadChannel>, Serializable {
+
+    private static final long serialVersionUID = 3889420316004453706L;
+
+    private final StorageOptions serviceOptions;
+    private final BlobId blob;
+    private final Map<StorageRpc.Option, ?> requestOptions;
+    private final int position;
+    private final boolean isOpen;
+    private final boolean endOfStream;
+    private final int chunkSize;
+
+    StateImpl(Builder builder) {
+      this.serviceOptions = builder.serviceOptions;
+      this.blob = builder.blob;
+      this.requestOptions = builder.requestOptions;
+      this.position = builder.position;
+      this.isOpen = builder.isOpen;
+      this.endOfStream = builder.endOfStream;
+      this.chunkSize = builder.chunkSize;
+    }
+
+    static class Builder {
+      private final StorageOptions serviceOptions;
+      private final BlobId blob;
+      private final Map<StorageRpc.Option, ?> requestOptions;
+      private int position;
+      private boolean isOpen;
+      private boolean endOfStream;
+      private int chunkSize;
+
+      private Builder(StorageOptions options, BlobId blob, Map<StorageRpc.Option, ?> reqOptions) {
+        this.serviceOptions = options;
+        this.blob = blob;
+        this.requestOptions = reqOptions;
+      }
+
+      Builder position(int position) {
+        this.position = position;
+        return this;
+      }
+
+      Builder isOpen(boolean isOpen) {
+        this.isOpen = isOpen;
+        return this;
+      }
+
+      Builder endOfStream(boolean endOfStream) {
+        this.endOfStream = endOfStream;
+        return this;
+      }
+
+      Builder chunkSize(int chunkSize) {
+        this.chunkSize = chunkSize;
+        return this;
+      }
+
+      RestorableState<BlobReadChannel> build() {
+        return new StateImpl(this);
+      }
+    }
+
+    static Builder builder(
+        StorageOptions options, BlobId blob, Map<StorageRpc.Option, ?> reqOptions) {
+      return new Builder(options, blob, reqOptions);
+    }
+
+    @Override
+    public BlobReadChannel restore() {
+      BlobReadChannelImpl channel = new BlobReadChannelImpl(serviceOptions, blob, requestOptions);
+      channel.position = position;
+      channel.isOpen = isOpen;
+      channel.endOfStream = endOfStream;
+      channel.chunkSize = chunkSize;
+      return channel;
+    }
+
+    @Override
+    public int hashCode() {
+      return Objects.hash(serviceOptions, blob, requestOptions, position, isOpen, endOfStream,
+          chunkSize);
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+      if (obj == null) {
+        return false;
+      }
+      if (!(obj instanceof StateImpl)) {
+        return false;
+      }
+      final StateImpl other = (StateImpl) obj;
+      return Objects.equals(this.serviceOptions, other.serviceOptions) &&
```

Comment:
the preferred style is to wrap before the '&&'. see [here](http://google.github.io/styleguide/javaguide.html#s4.5.1-line-wrapping-where-to-break).

```java
 return Objects.equals(this.serviceOptions, other.serviceOptions)
    && Objects.equals(this.blob, other.blob)
    && ...
```